lib/Mojo/Message/Request.pm

```diff
@@ -72,6 +72,9 @@ sub fix_headers {
     $headers->authorization('Basic ' . b64_encode($info, ''));
   }
 
+  # RFC 7230 3.3.2
+  $headers->remove('Content-Length') if ($headers->content_length // '') eq '0';
```

Comment:
I agree with @forki that you should also check if it's a GET request. And why do you do `// ''` instead of `// 0`?